packages/block-library/src/pullquote/edit.js

```diff
@@ -33,9 +33,12 @@ class PullQuoteEdit extends Component {
 	}
 
 	pullQuoteMainColorSetter( colorValue ) {
-		const { colorUtils, textColor, setTextColor, setMainColor } = this.props;
+		const { colorUtils, textColor, setTextColor, setMainColor, className } = this.props;
+		const isSolidColorStyle = includes( className, SOLID_COLOR_CLASS );
+		const shouldSetTextColor = isSolidColorStyle && ! textColor.color && colorValue;
+
 		setMainColor( colorValue );
-		if ( ! textColor.color || this.wasTextColorAutomaticallyComputed ) {
+		if ( shouldSetTextColor || this.wasTextColorAutomaticallyComputed ) {
```

Comment:
If we go the solid color style we pick the main color, and then we change to the regular style, each time we change the main color the text color continue to change.
I think maybe we can solve this by updating shouldSetTextColor to be `isSolidColorStyle && colorValue && ( ! textColor.color || this.wasTextColorAutomaticallyComputed );`

Reply:
Ah yes, nice! Updated